openxr/src/passthrough.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{Result, Graphics, Session, sys, cvt};
+use std::ptr;
+
+/// TODO Doc
+/// TODO Link https://developer.oculus.com/documentation/native/android/mobile-passthrough/
+/// TODO Pause/unpause
+/// Create the passthrough feature.
+///
+/// Requires [`XR_FB_passthrough`](https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_FB_passthrough)
+/// to be enabled.
+///
+/// See:
+/// * [`XR_FB_passthrough`](openxr/src/generated.rs:62)
+pub fn create_passthrough_fb<G: Graphics>(session: &Session<G>, flags: sys::PassthroughFlagsFB) -> Result<sys::PassthroughFB> {
+    let info = sys::PassthroughCreateInfoFB {
+        ty: sys::PassthroughCreateInfoFB::TYPE,
+        next: ptr::null(),
+        flags
+    };
+    let ext = session
+        .instance()
+        .exts()
+        .fb_passthrough
+        .as_ref()
+        .expect("FB_passthrough not loaded");
+    let mut passthrough = sys::PassthroughFB::NULL;
+    // TODO Implement a Passthrough struct & drop?
+    unsafe {
+        cvt((ext.create_passthrough)(session.as_raw(), &info, &mut passthrough))?;
+    }
+    Ok(passthrough)
+}
+
+/// TODO Doc
+/// TODO Link https://developer.oculus.com/documentation/native/android/mobile-passthrough/
+/// TODO More params: purpose, flags
+/// TODO Pause/unpause
+pub fn create_passthrough_layer<G: Graphics>(session: &Session<G>, passthrough: sys::PassthroughFB) -> Result<sys::PassthroughLayerFB> {
+    let info = sys::PassthroughLayerCreateInfoFB {
+        ty: sys::PassthroughLayerCreateInfoFB::TYPE,
+        next: ptr::null(),
+        passthrough,
+        flags: sys::PassthroughFlagsFB::IS_RUNNING_AT_CREATION,
+        purpose: sys::PassthroughLayerPurposeFB::RECONSTRUCTION
+    };
+    let ext = session
+        .inner
+        .instance
+        .exts()
+        .fb_passthrough
+        .as_ref()
+        .expect("FB_passthrough not loaded");
+    let mut layer = sys::PassthroughLayerFB::NULL;
+    // TODO Implement a Passthrough struct & drop?
+    unsafe {
+        cvt((ext.create_passthrough_layer)(session.as_raw(), &info, &mut layer))?;
+    }
+    Ok(layer)
+}
+
+/// TODO Doc
+/// TODO Link https://developer.oculus.com/documentation/native/android/mobile-passthrough/
+/// TODO Params: flags
+/// TODO Composition
+pub fn new_passthrough_composition_layer(layer: sys::PassthroughLayerFB) -> sys::CompositionLayerPassthroughFB {
```

Comment:
This should be exposed through a generated builder similar to `CompositionLayerProjection`.

Reply:
I removed this change and left the `CompositionLayerPassthroughFB` untouched. I played with the generator a bit but couldn't get satisfying results so far. Leaving this as-is doesn't prevent enabling & using passthrough, and I fear the required changes may make this PR grow a bit too much. I'll save this for later if you don't mind.